aten/src/ATen/ParallelNative.cpp

```diff
@@ -116,14 +116,29 @@ bool in_parallel_region() {
 
 void intraop_launch(std::function<void()> func) {
   if (!in_parallel_region() && get_num_threads() > 1) {
-    internal::_get_intraop_pool().run([func](){
-      func();
-    });
+    internal::_get_intraop_pool().run(func);
   } else {
     // execute inline if we're in parallel region
     func();
   }
+}
 
+std::future<void> intraop_launch_future(std::function<void()> func) {
+  auto func_promise = std::make_shared<std::promise<void>>();
```

Comment:
std::function requires the functor to be copyable, so, unfortunately, we cannot just move std::promise into lambda